pkg/ebpf/bytecode/runtime/runtime_asset.go

```diff
@@ -38,15 +39,23 @@ type CompiledOutput interface {
 
 // RuntimeAsset represents an asset that needs its content integrity checked at runtime
 type RuntimeAsset struct {
-	filename string
-	hash     string
+	filename  string
+	hash      string
+	telemetry Telemetry
 }
 
 func NewRuntimeAsset(filename, hash string) *RuntimeAsset {
-	return &RuntimeAsset{
-		filename: filename,
-		hash:     hash,
+	asset := &RuntimeAsset{
+		filename:  filename,
+		hash:      hash,
+		telemetry: newNoOpTelemetry(),
 	}
+
+	if filename == "tracer.c" || filename == "runtime-security.c" {
```

Comment:
I'm not sure you need this check. All the runtime assets are intended to be compiled.